apisix/admin/init.lua

```diff
@@ -256,23 +238,71 @@ local function reload_plugins(data, event, source, pid)
 end
 
 
+-- /apisix/admin/routes
+-- /apisix/admin/routes/1
+-- /apisix/admin/http/routes
+-- /apisix/admin/http/routes/1
+-- /apisix/admin/:workspace/http/routes
+-- /apisix/admin/:workspace/http/routes/1
+local valid_models = {
+    http = true,
+    stream = true,
+}
+
+local function run()
+    local api_ctx = {}
+    core.ctx.set_vars_meta(api_ctx)
+
+    local ok, err = check_token(api_ctx)
+    if not ok then
+        core.log.warn("failed to check token: ", err)
+        core.response.exit(401)
+    end
+
+    local uri_segs = core.utils.split_uri(ngx.var.uri)
+    core.log.info("uri: ", core.json.delay_encode(uri_segs))
+
+    local typ = uri_segs[4]
+    if resources_http[typ] then
+        -- /apisix/admin/routes
+        uri_segs = {"", "apisix", "admin", "", "http", unpack_tab(uri_segs, 4)}
+
+    elseif valid_models[typ] then
+        -- /apisix/admin/http/routes
+        -- /apisix/admin/stream/routes/1
+        uri_segs = {"", "apisix", "admin", "", unpack_tab(uri_segs, 4)}
+
+    else
+        -- /apisix/admin/:workspace/http/routes
```

Comment:
maybe we should add test case for this uri style ?

Reply:
good catch, I will add test cases later

Reply:
fixed already, please take a look when you have time